sdk/eventhub/Azure.Messaging.EventHubs/src/Authorization/EventHubSharedKeyCredential.cs

```diff
@@ -0,0 +1,90 @@
+        ///
+        public EventHubSharedKeyCredential(string sharedAccessKeyName,
+                                           string sharedAccessKey)
+        {
+            Guard.ArgumentNotNullOrEmpty(nameof(sharedAccessKeyName), sharedAccessKeyName);
+            Guard.ArgumentNotNullOrEmpty(nameof(sharedAccessKey), sharedAccessKey);
+
+            SharedAccessKeyName = sharedAccessKeyName;
+            SharedAccessKey = sharedAccessKey;
+        }
+
+        /// <summary>
+        ///   Retrieves the token that represents the shared access signature credential, for
+        ///   use in authorization against an Event Hub.
+        /// </summary>
+        ///
+        /// <param name="scopes">The access scopes to request a token for.</param>
+        /// <param name="cancellationToken">The token used to request cancellation of the operation.</param>
+        ///
+        /// <returns>The token representating the shared access signature for this credential.</returns>
+        ///
+        public override AccessToken GetToken(string[] scopes, CancellationToken cancellationToken) => throw new InvalidOperationException(Resources.SharedKeyCredentialCannotGenerateTokens);
+
+        /// <summary>
+        ///   Retrieves the token that represents the shared access signature credential, for
+        ///   use in authorization against an Event Hub.
+        /// </summary>
+        ///
+        /// <param name="scopes">The access scopes to request a token for.</param>
+        /// <param name="cancellationToken">The token used to request cancellation of the operation.</param>
+        ///
+        /// <returns>The token representating the shared access signature for this credential.</returns>
+        ///
+        public override Task<AccessToken> GetTokenAsync(string[] scopes, CancellationToken cancellationToken) => throw new InvalidOperationException(Resources.SharedKeyCredentialCannotGenerateTokens);
+
+        /// <summary>
+        /// Coverts to shared access signature credential.
+        /// </summary>
+        ///
+        /// <param name="eventHubResource">The Event Hubs resource to which the token is intended to serve as authorization.</param>
+        /// <param name="signatureValidityDuration">The duration that the signature should be considered valid; if not specified, a default will be assumed.</param>
+        ///
+        /// <returns>A <see cref="SharedAccessSignatureCredential" /> based on the requested shared access key.</returns>
+        ///
+        internal SharedAccessSignatureCredential CovertToSharedAccessSignatureCredential(string eventHubResource,
+                                                                                         TimeSpan? signatureValidityDuration = default) =>
+            new SharedAccessSignatureCredential(new SharedAccessSignature(eventHubResource, SharedAccessKeyName, SharedAccessKey, signatureValidityDuration));
+
```

Comment:
nit: blank new line

Reply:
That's intentional.   Convention is that the lambda syntax comes without a blank (it's intended to be single-line, just wrapped due to length.)